src/core/etl/src/Flow/ETL/Row/Entry/DateTimeEntry.php

```diff
@@ -16,14 +16,22 @@ final class DateTimeEntry implements \Stringable, Entry
 {
     use EntryRef;
 
+    private readonly \DateTimeInterface $value;
+
     /**
      * @throws InvalidArgumentException
      */
-    public function __construct(private readonly string $name, private readonly \DateTimeInterface $value)
+    public function __construct(private readonly string $name, \DateTimeInterface|string $value)
     {
         if ($name === '') {
             throw InvalidArgumentException::because('Entry name cannot be empty');
         }
+
+        if (\is_string($value)) {
+            $this->value = new \DateTimeImmutable($value);
```

Comment:
there should be a try-catch here in case someone will pass a value that is not a valid datetime string so we can throw consistent InvalidArgumentException

Reply:
Added also into json entry.

Reply:
Please add at least one unit tests for it, I would like to see how the exception message is going to look